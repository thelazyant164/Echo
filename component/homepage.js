import {React,useState} from 'react';
import { StyleSheet, Text, View,FlatList } from 'react-native';
import { Featurebutton } from './page-component/feature-button';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { Header } from './page-component/header';
const Stack = createNativeStackNavigator();
export const Homepage =({navigation})=>{

    const [data,setData]= useState(["Home","Files storage"])
    const allfeature =["Noise cancelling", "Volume adjust","Speech to text","Noise reduction record"];
    return(
        <View>
        <Header></Header>
        <View style={{marginTop:60 }}>
            
                <Text style={{textAlign: "center",fontSize:20}}>Home</Text>
                <View style={style.feature_container}>
                <Text>Recently</Text>
                <View style={style.container}>
                <FlatList
                    numColumns={4}
                    
                    data={data}
                    renderItem={
                    ({item,key}) => 
                    {
                    return(
                        <Featurebutton feature={item} navigation={navigation}></Featurebutton>
                    )
                    }
                }
               
                />
                </View>
            </View>

            <View style={style.feature_container}>
                <Text>Feature</Text>
                <View style={style.container}>
                    <FlatList
                    numColumns={4}
                        data={allfeature}
                        renderItem={
                        ({item,key}) => 
                        {
                        return(
                            <Featurebutton feature={item} navigation={navigation}></Featurebutton>
                        )
                        }
                    }
                    >

                    </FlatList>

                </View>

            </View>
        </View>
        </View>
    )
}


const style=StyleSheet.create({
    feature_container:{
        marginTop:100,
        marginLeft:10,
        marginRight:10
        
    },
    container: {
        backgroundColor:"#F8F5F5",
        borderRadius:12,        
    },
    list:{
        flexDirection: "row",
        flexWrap: "wrap",
    },

});
