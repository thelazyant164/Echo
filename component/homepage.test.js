import { React } from 'react';
import renderer from 'react-test-renderer';
import { Homepage } from './homepage';

describe('<Homepage />', () => {
  let homepage;
  beforeEach(() => {
    homepage = renderer.create(<Homepage />).toJSON();
  });

  test('Renders correctly', () => {
    expect(homepage).toMatchSnapshot();
  });
});
