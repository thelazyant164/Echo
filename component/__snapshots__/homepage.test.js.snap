// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Homepage /> Renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#D9D9D9",
        "height": 60,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
          "justifyContent": "center",
          "marginLeft": 20,
          "marginTop": 10,
        }
      }
    >
      Echo
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 60,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      Home
    </Text>
    <View
      style={
        Object {
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 100,
        }
      }
    >
      <Text>
        Recently
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#F8F5F5",
            "borderRadius": 12,
          }
        }
      >
        <RCTScrollView
          data={
            Array [
              "Home",
              "Files storage",
            ]
          }
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#D9D9D9",
                      "borderRadius": 12,
                      "height": 90,
                      "marginBottom": "10px",
                      "marginLeft": 4,
                      "marginRight": 4,
                      "marginTop": "10px",
                      "opacity": 1,
                      "width": 90,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "justifyContent": "center",
                        "margin": "3px",
                        "textAlign": "center",
                      }
                    }
                  >
                    <Text />
                    <Text>
                      Home
                    </Text>
                  </View>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#D9D9D9",
                      "borderRadius": 12,
                      "height": 90,
                      "marginBottom": "10px",
                      "marginLeft": 4,
                      "marginRight": 4,
                      "marginTop": "10px",
                      "opacity": 1,
                      "width": 90,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "justifyContent": "center",
                        "margin": "3px",
                        "textAlign": "center",
                      }
                    }
                  >
                    <Text />
                    <Text>
                      Files storage
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
    <View
      style={
        Object {
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 100,
        }
      }
    >
      <Text>
        Feature
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#F8F5F5",
            "borderRadius": 12,
          }
        }
      >
        <RCTScrollView
          data={
            Array [
              "Noise cancelling",
              "Volume adjust",
              "Speech to text",
              "Noise reduction record",
            ]
          }
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#D9D9D9",
                      "borderRadius": 12,
                      "height": 90,
                      "marginBottom": "10px",
                      "marginLeft": 4,
                      "marginRight": 4,
                      "marginTop": "10px",
                      "opacity": 1,
                      "width": 90,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "justifyContent": "center",
                        "margin": "3px",
                        "textAlign": "center",
                      }
                    }
                  >
                    <Text />
                    <Text>
                      Noise cancelling
                    </Text>
                  </View>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#D9D9D9",
                      "borderRadius": 12,
                      "height": 90,
                      "marginBottom": "10px",
                      "marginLeft": 4,
                      "marginRight": 4,
                      "marginTop": "10px",
                      "opacity": 1,
                      "width": 90,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "justifyContent": "center",
                        "margin": "3px",
                        "textAlign": "center",
                      }
                    }
                  >
                    <Text />
                    <Text>
                      Volume adjust
                    </Text>
                  </View>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#D9D9D9",
                      "borderRadius": 12,
                      "height": 90,
                      "marginBottom": "10px",
                      "marginLeft": 4,
                      "marginRight": 4,
                      "marginTop": "10px",
                      "opacity": 1,
                      "width": 90,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "justifyContent": "center",
                        "margin": "3px",
                        "textAlign": "center",
                      }
                    }
                  >
                    <Text />
                    <Text>
                      Speech to text
                    </Text>
                  </View>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#D9D9D9",
                      "borderRadius": 12,
                      "height": 90,
                      "marginBottom": "10px",
                      "marginLeft": 4,
                      "marginRight": 4,
                      "marginTop": "10px",
                      "opacity": 1,
                      "width": 90,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "justifyContent": "center",
                        "margin": "3px",
                        "textAlign": "center",
                      }
                    }
                  >
                    <Text />
                    <Text>
                      Noise reduction record
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</View>
`;
